### PR TITLE
Record processed_from_queue stat

### PR DESCRIPTION
**Pull request by brandon-m-skinner**
## Description

Added a stat to keep track of when things are processed from the queue for an index.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [x] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Run StatsD locally
2. Add somewhere it will run `define( 'VIP_STATSD_HOST', 'docker.for.mac.localhost' );`
3. Add somewhere it will run `define( 'VIP_STATSD_PORT', 8125 );`
4. Apply PR.
5. Ensure VIP Search is enabled and the `wp_vip_search_index_queue` DB table and related functionality works locally.
6. Add a bunch of entries to the `wp_vip_search_index_queue` DB table.
7. `wp shell`
8. Build an array of assoc arrays where 'job_id' is equal to the object ids of the entries you added to `wp_vip_search_index_queue`.
9. Call the `Queue::process_jobs` function on the array of assoc arrays you built in `wp shell` previously.
10. The number of jobs you processed should equal the stat from the StatsD output. Should be something lilke `com.wordpress.elasticsearch.unknown.unknown.processed_from_queue.200508.vip-200508-post-1`.

### CODE REVIEW

File: search/includes/classes/class-queue.php [hunk 1]
```diff
@@ -471,10 +471,36 @@ public function process_jobs( $jobs ) {
...
 		}
 	}
 
+	public function record_processed_from_queue_stat( $count, $indexable ) {
+		if ( ! $indexable ) {
+			return;
+		}
+
+		if ( ! is_int( $count ) ) {
+			$count = intval( $count );
```
Comment (nickdaugherty) [inline search/includes/classes/class-queue.php:486]: Should this instead return? If a non-int came in, we probably don't want to rely on PHP typecasting for stat values.

Comment (brandon-m-skinner) [reply search/includes/classes/class-queue.php:486]: I don't think it makes much of a difference. Last time we had the same conversation on different sides 😄.

[intval on invalid values return 0](https://www.php.net/manual/en/function.intval.php#refsect1-function.intval-returnvalues). Incrementing by 0 doesn't do anything, so I think it's fine.

Comment (nickdaugherty) [reply search/includes/classes/class-queue.php:486]: Alright, LGTM then

**Comment by brandon-m-skinner** [comment]
r1017-stacks

**Pull request merged**